modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -1534,18 +1551,31 @@ def test_reset_index_with_named_index(index_levels_names_max_levels):
     )
     modin_df.index.name = pandas_df.index.name = index_name
     df_equals(modin_df, pandas_df)
+    if test_async_reset_index:
+        # The change in index is not automatically handled by Modin.
+        modin_df._query_compiler._modin_frame._propagate_index_objs(axis=0)
```

Comment:
Don't we want to call `_propagate_index_objs` even if we change index.name?

Reply:
This is how it works now. Or do you mean that regardless of the value of `test_async_reset_index` variable?

Reply:
I mean in general. If we called `_propagate_index_objs` only in the tests regardless of the value of `test_async_reset_index` variable, we would hide possible errors outside of the tests. Does `_propagate_index_objs` really introduce significant slowdown for further operations when doing `modin_df.index.name = index_name`?

Reply:
I can only say that this will especially affect dataframes with complex data types, for which zero copy serialization will not work. Impact research should be done as part of the work with https://github.com/modin-project/modin/issues/3941, I guess :)

Reply:
I would like to see the issue to be mentioned in this in-code comment then, explaining that this is a hack to workaround the known problem